src/collector/filter_collector_wrapper.rs

```diff
@@ -0,0 +1,154 @@
+// # Custom collector example
+//
+// This example shows how you can implement your own
+// collector. As an example, we will compute a collector
+// that computes the standard deviation of a given fast field.
+//
+// Of course, you can have a look at the tantivy's built-in collectors
+// such as the `CountCollector` for more examples.
+
+// ---
+// Importing tantivy...
+use std::marker::PhantomData;
+
+use crate::collector::{Collector, SegmentCollector};
+use crate::fastfield::FastFieldReader;
+use crate::schema::Field;
+use crate::{Score, SegmentReader, TantivyError};
+
+/// The `FilterCollector` collector filters docs using a u64 fast field value and a predicate.
+/// Only the documents for which the predicate returned "true" will be passed on to the next collector.
+///
+/// ```rust
+/// use tantivy::collector::{TopDocs, FilterCollector};
+/// use tantivy::query::QueryParser;
+/// use tantivy::schema::{Schema, TEXT, INDEXED, FAST};
+/// use tantivy::{doc, DocAddress, Index};
+///
+/// let mut schema_builder = Schema::builder();
+/// let title = schema_builder.add_text_field("title", TEXT);
+/// let price = schema_builder.add_u64_field("price", INDEXED | FAST);
+/// let schema = schema_builder.build();
+/// let index = Index::create_in_ram(schema);
+///
+/// let mut index_writer = index.writer_with_num_threads(1, 10_000_000).unwrap();
+/// index_writer.add_document(doc!(title => "The Name of the Wind", price => 30_200u64));
+/// index_writer.add_document(doc!(title => "The Diary of Muadib", price => 29_240u64));
+/// index_writer.add_document(doc!(title => "A Dairy Cow", price => 21_240u64));
+/// index_writer.add_document(doc!(title => "The Diary of a Young Girl", price => 20_120u64));
+/// assert!(index_writer.commit().is_ok());
+///
+/// let reader = index.reader().unwrap();
+/// let searcher = reader.searcher();
+///
+/// let query_parser = QueryParser::for_index(&index, vec![title]);
+/// let query = query_parser.parse_query("diary").unwrap();
+/// let no_filter_collector = FilterCollector::new(price, &|value| true, TopDocs::with_limit(2));
+/// let top_docs = searcher.search(&query, &no_filter_collector).unwrap();
+///
+/// assert_eq!(top_docs.len(), 2);
+/// assert_eq!(top_docs[0].1, DocAddress(0, 1));
+/// assert_eq!(top_docs[1].1, DocAddress(0, 3));
+///
+/// let filter_all_collector = FilterCollector::new(price, &|value| false, TopDocs::with_limit(2));
+/// let filtered_top_docs = searcher.search(&query, &filter_all_collector).unwrap();
+///
+/// assert_eq!(filtered_top_docs.len(), 0);
+/// ```
+pub struct FilterCollector<TCollector, TSegmentCollector> {
+    field: Field,
+    collector: TCollector,
+    predicate: &'static (dyn Fn(u64) -> bool + Send + Sync),
+    phantom: PhantomData<TSegmentCollector>,
+}
+
+impl<TCollector, TSegmentCollector> FilterCollector<TCollector, TSegmentCollector>
+where
+    TCollector: Collector<Child = TSegmentCollector> + Send + Sync,
+    TSegmentCollector: SegmentCollector + Send + Sync,
```

Comment:
`+ Send + Sync ` is not necessary for the TSegmentCollector I think

Reply:
https://github.com/tantivy-search/tantivy/commit/4e93c681c0c4ac0b83fc85f4fee09bbe27fb15c8